### PR TITLE
Added some Events

### PR DESCRIPTION
**Pull request by blackassasin1234**
adds PlayerBanEvent, PlayerUnbanEvent, PlayerIpBanEvent, and PlayerIpUnbanEvent..

### CODE REVIEW

File: core/src/io/anuke/mindustry/game/EventType.java [hunk 1]
```diff
@@ -327,11 +327,40 @@ public PlayerConnect(Player player){
...
+
         public PlayerLeave(Player player){
             this.player = player;
         }
     }
-           
+    public static class PlayerBanEvent{
+        public final Player player;
+
+        public PlayerBanEvent(Player player) {
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/game/EventType.java]: Unnecessary space, use the formatting guidelines. Happens for every method here.

File: core/src/io/anuke/mindustry/game/EventType.java [hunk 2]
```diff
@@ -327,11 +327,40 @@ public PlayerConnect(Player player){
...
+            this.player = player;
+        }
+    }
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/game/EventType.java]: Add a blank line after each class.

File: core/src/io/anuke/mindustry/net/Administration.java [hunk 3]
```diff
@@ -126,7 +131,7 @@ public boolean unbanPlayerID(String id){
         info.banned = false;
         bannedIPs.removeAll(info.ips, false);
         save();
-
+        Events.fire(new PlayerUnbanEvent(Vars.playerGroup.find(p -> p.uuid.equals(id))));
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/net/Administration.java]: `player.uuid` can be null, causing a crash here. The condition should be inverted.

File: core/src/io/anuke/mindustry/net/Administration.java [hunk 4]
```diff
@@ -88,7 +91,7 @@ public boolean banPlayerID(String id){
         getCreateInfo(id).banned = true;
 
         save();
-
+        Events.fire(new PlayerBanEvent(Vars.playerGroup.find(p -> p.uuid.equals(id))));
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/net/Administration.java]: Same issue with uuid.

File: core/src/io/anuke/mindustry/net/Administration.java (hunk 3 above)
Comment (blackassasin1234) [reply core/src/io/anuke/mindustry/net/Administration.java]: What do you mean by inverting the condition?

Comment (Anuken) [reply core/src/io/anuke/mindustry/net/Administration.java]: `id.equals(p.uuid)` instead of `p.uuid.equals(id)`

**Comment by blackassasin1234** [comment]
Formatted the code and used 
id.equals(p.uuid) instead of p.uuid.equals(id)

**Pull request merged**